src/kernel/arch/x86/arch.zig

```diff
@@ -251,6 +254,31 @@ pub fn haltNoInterrupts() noreturn {
     }
 }
 
+///
+/// Write a byte to serial port com1. Used by the serial initialiser
+///
+/// Arguments:
+///     IN byte: u8 - The byte to write
+///
+fn writeSerialCom1(byte: u8) void {
+    x86_serial.write(byte, x86_serial.Port.COM1);
+}
```

Comment:
Then this can be used for the Serial.init(), so be passed the function: `Serial.init(writeSerialCom1);` So this will be exposed as `pub`.

Reply:
The thing is that the architecture is what decides how serial is done. Not all architectures will have the equivalent of a writeSerialCom1 function so it's best left to the arch to organise.